### PR TITLE
Added API and SSL

### PR DESCRIPTION
**Pull request by arturek**
No real operations are perfomed although API should work correctly. Validation and many details not implemented yet.

### CODE REVIEW

File: src/Bullfrog.Api/EswSslExtentions.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                     listenOptions.NoDelay = true;
+                 });
+             });
+        }
+
+        private static X509Certificate2 GetCertificate(string environment)
+        {
+            var tld = "PROD".Equals(environment, StringComparison.OrdinalIgnoreCase) ? "com" : "net";
+            var subject = $"star.{environment}.eshopworld.{tld}";
+            var store = new X509Store(StoreName.My, StoreLocation.LocalMachine);
```
Comment (VaclavK) [inline src/Bullfrog.Api/EswSslExtentions.cs]: have we considered KV?

File: src/Bullfrog.Actor/Program.cs [hunk 2]
```diff
@@ -1,34 +1,44 @@
...
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.ServiceFabric.Actors.Runtime;
+using Autofac;
+using Autofac.Integration.ServiceFabric;
+using Bullfrog.Common.DependencyInjection;
+using Castle.Core.Internal;
+using Eshopworld.Telemetry;
+
+[assembly: InternalsVisibleTo(InternalsVisible.ToDynamicProxyGenAssembly2)]
```
Comment (dguerin) [inline src/Bullfrog.Actor/Program.cs]: Could we put this into assembly info? Bit nasty to have it out here.

File: src/Bullfrog.Api/ApiService.cs [hunk 3]
```diff
@@ -49,6 +49,7 @@ protected override IEnumerable<ServiceInstanceListener> CreateServiceInstanceLis
                                 .UseStartup<Startup>()
                                 .UseServiceFabricIntegration(listener, ServiceFabricIntegrationOptions.None)
                                 .UseUrls(url)
+                                .UseEswSsl(listener.ServiceContext.CodePackageActivationContext.GetEndpoint("BullfrogApiEndpoint").Port)
```
Comment (dguerin) [inline src/Bullfrog.Api/ApiService.cs]: Could we do something a little cleaner here and hide the port reference within the extension method? We might be able to do something with the SF Template here.

File: src/Bullfrog.Api/Controllers/BaseManagementController.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                actorName = actorName.Substring(1);
+            var actorId = new ActorId($"{actorName}:{scaleGroup}/{region}");
+            var serviceName = actorServices[typeof(TActor)];
+            var actorUri = new Uri($"{StatelessServiceContext.CodePackageActivationContext.ApplicationName}/{serviceName}");
+            return ActorProxy.Create<TActor>(actorId, actorUri);
+        }
+
+        protected string[] ListRegionsOfScaleGroup(string scaleGroup)
+        {
+            switch (scaleGroup)
```
Comment (dguerin) [inline src/Bullfrog.Api/Controllers/BaseManagementController.cs]: Might want to do some casing protection here? Ie to upper or lower

File: src/Bullfrog.Api/EswSslExtentions.cs (hunk 1 above)
Comment (dguerin) [reply src/Bullfrog.Api/EswSslExtentions.cs]: Yeah I would have though we could grab them from keyvault when the MSI of the scale set is auth to the keyvault to get the cert

File: src/Bullfrog.Api/Controllers/BaseManagementController.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            var actorName = typeof(TActor).Name;
+            if (actorName.StartsWith('I'))
+                actorName = actorName.Substring(1);
+            var actorId = new ActorId($"{actorName}:{scaleGroup}/{region}");
+            var serviceName = actorServices[typeof(TActor)];
+            var actorUri = new Uri($"{StatelessServiceContext.CodePackageActivationContext.ApplicationName}/{serviceName}");
+            return ActorProxy.Create<TActor>(actorId, actorUri);
+        }
+
+        protected string[] ListRegionsOfScaleGroup(string scaleGroup)
```
[the remark this answers is not in the document]
Comment (arturek) [reply src/Bullfrog.Api/Controllers/BaseManagementController.cs]: I'll agree it should be configurable somewhere else. This method is here only temporarily.

File: src/Bullfrog.Api/EswSslExtentions.cs (hunk 1 above)
Comment (arturek) [reply src/Bullfrog.Api/EswSslExtentions.cs]: It's done currently this way. Changes probably require some more discussion.

File: src/Bullfrog.Actor/Program.cs (hunk 2 above)
Comment (arturek) [reply src/Bullfrog.Actor/Program.cs]: I'll move it. But there's no assembly info file anymore...

File: src/Bullfrog.Api/ApiService.cs (hunk 3 above)
Comment (arturek) [reply src/Bullfrog.Api/ApiService.cs]: We can think about it when working on middleware. Than probably we could handle the most frequent case with one endpoint automatically.

**Pull request merged**